tcs-service/src/test/java/com/transformuk/hee/tis/tcs/service/service/impl/ProgrammeMembershipServiceImplTest.java

```diff
@@ -597,4 +609,54 @@ public void shouldDeleteCurriculumMemberships() {
     verify(applicationEventPublisherMock, times(2)).publishEvent(any(
         CurriculumMembershipDeletedEvent.class));
   }
+
+  @Test
+  public void shouldReturnDtoWithErrorWhenIdNotFound() {
+    ProgrammeMembershipDTO dto = new ProgrammeMembershipDTO();
+    dto.setUuid(PROGRAMME_MEMBERSHIP_ID_1);
+
+    when(programmeMembershipRepositoryMock.findByUuid(PROGRAMME_MEMBERSHIP_ID_1))
+        .thenReturn(Optional.empty());
+
+    ProgrammeMembershipDTO returnDto = testObj.patch(dto);
+    Assert.assertEquals(1, returnDto.getMessageList().size());
+    Assert.assertEquals("Programme membership id not found.",
+        returnDto.getMessageList().get(0));
+  }
+
+  @Test
+  public void shouldReturnDtoWithErrorWhenPatchPmValidationFails() {
+    ProgrammeMembershipDTO dto = new ProgrammeMembershipDTO();
+    dto.setUuid(PROGRAMME_MEMBERSHIP_ID_1);
+
+    when(programmeMembershipRepositoryMock.findByUuid(PROGRAMME_MEMBERSHIP_ID_1))
+        .thenReturn(Optional.of(programmeMembership1));
+    lenient().doNothing().when(programmeMembershipValidatorMock).validateForBulk(argThat(
+        new ArgumentMatcher<ProgrammeMembershipDTO>() {
+          @Override
+          public boolean matches(ProgrammeMembershipDTO programmeMembershipDto) {
+            programmeMembershipDto.addMessage("default error");
+            return false;
+          }
+        }));
+
+    ProgrammeMembershipDTO returnDto = testObj.patch(dto);
+    Assert.assertEquals(1, returnDto.getMessageList().size());
+  }
+
+  @Test
+  public void shouldPatchProgrammeMembership() {
+    ProgrammeMembershipDTO dto = new ProgrammeMembershipDTO();
+    dto.setUuid(PROGRAMME_MEMBERSHIP_ID_1);
+
+    when(programmeMembershipRepositoryMock.findByUuid(PROGRAMME_MEMBERSHIP_ID_1))
+        .thenReturn(Optional.of(programmeMembership1));
+    doNothing().when(programmeMembershipValidatorMock).validateForBulk(any());
```

Comment:
This looks like it should be a `verify()` with the assertion

Reply:
Thanks, it's done.